### PR TITLE
Button component

### PR DESCRIPTION
**Pull request by HIELIYA2**

### CODE REVIEW

File: src/cmps/button/Button.css [hunk 1]
```diff
@@ -0,0 +1,58 @@
+
+.button{
+    min-width: 130px;
```
Comment (saracodes) [inline src/cmps/button/Button.css]: mmm, I didn't mention the rem unit.
start with reading and understanding: https://snook.ca/archives/html_and_css/font-size-with-rem

later on we will add a ticket for refactoring and removing all `px` units

File: src/cmps/button/Button.css [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    color: #D5C9FF;
+}
+.tertiary.disabled{
+    border: 0 solid rgb(139, 0, 139);
+    background-color: #F0EFF1;
+    color: #D4D3D4;
+}
+
+
+
```
Comment (saracodes) [inline src/cmps/button/Button.css]: too many empty lines (should be fixed automatically by prettier, pls check why not)

File: src/cmps/button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+//Button component
```
Comment (saracodes) [inline src/cmps/button/Button.tsx]: remove redundant comments
+in line 6

File: src/cmps/button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+const Button: React.FC<props> = ({ isDisabled, value, onClick: handleClick }) => {
+    var styleDisable: string = '';
+    if (isDisabled) { styleDisable = 'default' } else { styleDisable = 'disabled' };
+    const classes: string = `button ${value} ${styleDisable}`;
+    console.log(value);
+    console.log(styleDisable);
+    return (
+        <div className="buttons">
+            <button onClick={handleClick} className={classes}>{value}</button>
```
Comment (saracodes) [inline src/cmps/button/Button.tsx]: I'm pretty sure that I've seen a more updated version of this file (we refactored together the `handleClick`).
can you pls make sure you pushed all commits?

File: src/cmps/tag/Tag.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    onClick(e: MouseEvent<HTMLElement>): void;
+}
+
+const Tag: React.FC<props> = ({ isDeletable, value, onClick: handleClick }) => {
+    return (
+        <section>
+            <div className="tag-container">
+                <div className="tag-name">{value}</div>
+                <button onClick={handleClick} className="tag-x-button" >
+                    <FontAwesomeIcon icon="times" />
```
Comment (saracodes) [inline src/cmps/tag/Tag.tsx]: you can use the next ASCII character: ×
and TADA! no need for all `@ fortawesome` dependencies

File: src/cmps/tag/Tag.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+                </button>
+            </div>
+        </section>
+    )
+}
+
+export default Tag;
```
Comment (saracodes) [inline src/cmps/tag/Tag.tsx]: add empty line
(editorconfig should handle it with `insert_final_newline`, apparently, this file was created before we added editorconfig)

File: src/cmps/tags/Tags.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react";
+import './Tags.css';
+import Tag from '../tag/Tag';
+// eslint-disable-next-line
+import { string } from "prop-types";
```
Comment (saracodes) [inline src/cmps/tags/Tags.tsx]: mmm I think this line was added by mistake
wdyt?

File: src/cmps/tags/Tags.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            id: 2,
+            name: 'tag2'
+        }]
+    }
+
+    remove = () => {
+        console.log('button X clicked');
+    }
+
+    addTag = (event: any) => {
```
Comment (saracodes) [inline src/cmps/tags/Tags.tsx]: avoid using `any`, there is no point for using typescript this way 🤓

File: src/index.tsx [hunk 9]
```diff
@@ -4,7 +4,7 @@ import './assets/css/main.css';
 import App from './App';
 import * as serviceWorker from './serviceWorker';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+ReactDOM.render(< App />, document.getElementById('root'));
```
Comment (saracodes) [inline src/index.tsx]: redundant space was added

File: .prettierrc.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
+module.exports =  {
+  semi:  true,
+  trailingComma:  'all',
+  singleQuote:  true,
+  printWidth:  120,
+  tabWidth:  4,
```
[the remark this answers is not in the document]
Comment (HIELIYA2) [reply .prettierrc.js]: done

File: .prettierrc.js [hunk 11]
```diff
@@ -0,0 +1,7 @@
+module.exports =  {
+  semi:  true,
+  trailingComma:  'all',
+  singleQuote:  true,
+  printWidth:  120,
```
[the remark this answers is not in the document]
Comment (HIELIYA2) [reply .prettierrc.js]: I dont not see "max-len" rule in .eslintrc.js file...

File: src/App.tsx [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+      <Button value="secondary" isDisabled={false} onClick={() => console.log('4') }/>
+      <Button value="tertiary" isDisabled={true} onClick={() => console.log('5') }/>
+      <Button value="tertiary" isDisabled={false} onClick={()=>{}}/>
+      <Tags/>
+      </header>
+    </div>
+  );
+}
+
+export default App;
```
[the remark this answers is not in the document]
Comment (HIELIYA2) [reply src/App.tsx]: ok, I left it right now...

File: src/assets/css/app.css [hunk 13]
```diff
@@ -1,3 +1,18 @@
+* {
+  letter-spacing: 0.6px;
+  box-sizing: border-box;
+  color: black;
+}
+a{
```
[the remark this answers is not in the document]
Comment (HIELIYA2) [reply src/assets/css/app.css]: done

File: src/cmps/button/Button.css [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+    cursor: pointer;
+    font-weight: bold;
+    transition: 0.3s;
+    display: flex;
+    justify-content: center;
+}
+.primary.default{
+    border: 2px solid #7543F8;
+    background-color: #7543F8;
+    color: rgb(255, 255, 255)
```
[the remark this answers is not in the document]
Comment (HIELIYA2) [reply src/cmps/button/Button.css]: done

File: src/cmps/button/Button.css [hunk 15]
```diff
@@ -0,0 +1,58 @@
+
+.button{
+    min-width: 130px;
+    max-width: 130px;
+    letter-spacing: 3px;
```
[the remark this answers is not in the document]
Comment (HIELIYA2) [reply src/cmps/button/Button.css]: letter-spacing has been reset by a file reset.css so its equal to 0...

File: src/cmps/button/Button.css [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+    font-size: 14px;
+    font-family: Lato_regular;
```
[the remark this answers is not in the document]
Comment (HIELIYA2) [reply src/cmps/button/Button.css]: done

File: src/cmps/button/Button.css [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+    background-color: #D5C9FF;
+    color: rgb(255, 255, 255)
+}
+.secondary.disabled{
+    border: 2px solid #D5C9FF;
+    background-color: #fff;
+    color: #D5C9FF;
+}
+.tertiary.disabled{
+    border: 0 solid rgb(139, 0, 139);
```
[the remark this answers is not in the document]
Comment (HIELIYA2) [reply src/cmps/button/Button.css]: changed

File: src/cmps/button/Button.css (hunk 1 above)
Comment (HIELIYA2) [reply src/cmps/button/Button.css]: i know how to use in rem unit...

File: src/cmps/button/Button.css (hunk 2 above)
Comment (HIELIYA2) [reply src/cmps/button/Button.css]: fixed

File: src/cmps/button/Button.tsx (hunk 3 above)
Comment (HIELIYA2) [reply src/cmps/button/Button.tsx]: done

File: src/cmps/button/Button.tsx (hunk 4 above)
Comment (HIELIYA2) [reply src/cmps/button/Button.tsx]: you right... It has been modified already

File: src/cmps/tag/Tag.tsx (hunk 5 above)
Comment (HIELIYA2) [reply src/cmps/tag/Tag.tsx]: ok

File: src/cmps/tag/Tag.tsx (hunk 6 above)
Comment (HIELIYA2) [reply src/cmps/tag/Tag.tsx]: fix

File: src/cmps/tags/Tags.tsx (hunk 7 above)
Comment (HIELIYA2) [reply src/cmps/tags/Tags.tsx]: Not already relevant

File: src/cmps/tags/Tags.tsx (hunk 8 above)
Comment (HIELIYA2) [reply src/cmps/tags/Tags.tsx]: Do not know how to define event...

File: src/index.tsx (hunk 9 above)
Comment (HIELIYA2) [reply src/index.tsx]: fixed

File: .prettierrc.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (saracodes) [reply .prettierrc.js]: two small notes:
1. you made the changes on a branch which is not related to this PR (on `tags` and not on `button`) so I can't see in the PR all your changes 😓for next time...
2. the reviewer is the one who resolves the conversation so s/he makes sure it was handled as expected

File: .prettierrc.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (saracodes) [reply .prettierrc.js]: it is in the next format:
`"max-len": ["error", {"code": 120}]`

but in a second thought, you can leave it, so prettier will automatically fix it while the linter will warn if exists

File: src/cmps/button/Button.css (hunk 17 above)
[the remark this answers is not in the document]
Comment (saracodes) [reply src/cmps/button/Button.css]: I can still see it, have you pushed already?

File: src/cmps/button/Button.css (hunk 16 above)
[the remark this answers is not in the document]
Comment (saracodes) [reply src/cmps/button/Button.css]: I can still see it, have you pushed already?

File: src/cmps/tags/Tags.tsx (hunk 8 above)
Comment (saracodes) [reply src/cmps/tags/Tags.tsx]: me neither but Google knows:
`React.FormEvent<HTMLInputElement>`

Comment (saracodes) [reply src/cmps/tags/Tags.tsx]: https://stackoverflow.com/questions/42081549/typescript-react-event-types

**Pull request merged**